cwl/src/main/scala/cwl/model.scala

```diff
@@ -190,15 +190,16 @@ case class SchemaDefRequirement(
   def lookupCwlType(tpe: String): Option[SchemaDefTypes] = {
 
     def matchesType(inputEnumSchema: InputEnumSchema): Boolean = {
-      inputEnumSchema.name.fold(false){name => FileAndId(name)(ParentName.empty).id equalsIgnoreCase FullyQualifiedName(tpe)(ParentName.empty).id}
+      inputEnumSchema.name.fold(false){name => FileAndId(name)(ParentName.empty).id equalsIgnoreCase FileAndId(tpe)(ParentName.empty).id}
     }
 
     types.toList.flatMap {
-    case Inl(inputRecordSchema: InputRecordSchema) if FileAndId(inputRecordSchema.name)(ParentName.empty).id equalsIgnoreCase FullyQualifiedName(tpe)(ParentName.empty).id=>
-          List(Coproduct[SchemaDefTypes](inputRecordSchema))
-    case Inr(Inl(inputEnumSchema: InputEnumSchema)) if matchesType(inputEnumSchema)=>
-      List(Coproduct[SchemaDefTypes](inputEnumSchema))
-    case _ => List()
+      case Inl(inputRecordSchema: InputRecordSchema) if FileAndId(inputRecordSchema.name)(ParentName.empty).id equalsIgnoreCase FileAndId(tpe)(ParentName.empty).id =>
```

Comment:
[TOL] Not your fault but.....

I personally hate these `case Inl(...(Inl(Inr(value)...)` patterns because they're atrociously brittle to re-ordering of the coproduct definition. Much better IMO is the `select[InputRecordSchema]` method.

To use that method in a pattern match you'd need to define a custom unapply on the `SchemaDefTypes` coproduct with a case for each of its possible values.

Reply:
I wonder whether this is what makes IntelliJ hate this file so much [1]

[1] https://youtrack.jetbrains.com/issue/SCL-15243